lib/editor/components/timetable/TripSeriesModal.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+// $FlowFixMe: Flow doesn't know about useState: https://stackoverflow.com/questions/53105954/cannot-import-usestate-because-there-is-no-usestate-export-in-react-flow-with
+import React, {useState} from 'react'
+import {Button, Modal} from 'react-bootstrap'
+
+import * as tripActions from '../../actions/trip'
+import {getComponentMessages} from '../../../common/util/config'
+import HourMinuteInput from '../HourMinuteInput'
+import type {Trip} from '../../../types'
+
+type Props = {
+  addNewTrip: typeof tripActions.addNewTrip,
+  constructNewRow: (trip: ?Trip, tripSeriesStartTime: number) => ?Trip,
+  onClose: () => void,
+  show: boolean
+}
+
+const initialState = {endTime: 0, error: '', startTime: 0, headway: 0}
+
+const TripSeriesModal = (props: Props) => {
+  const [{endTime, error, headway, startTime}, setState] = useState(initialState)
+  const messages = getComponentMessages('TripSeriesModal')
+
+  const close = () => {
```

Comment:
This can probably be named `handleClose`! Although you don't need it at all if you get rid of the new state